client/src/pages/SignUp.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from 'react';
+import Joi from 'joi';
+
+import {
+  Avatar,
+  Button,
+  CssBaseline,
+  TextField,
+  Link,
+  Grid,
+  Box,
+  Typography,
+  Container,
+} from '@mui/material';
+import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
+import axios from 'axios';
+import BasicSelect from '../components';
+
+const Copyright = (props: any) => {
+  return (
+    <Typography
+      variant="body2"
+      color="text.secondary"
+      align="center"
+      {...props}
+    >
+      {'Copyright © '}
+      <Link color="inherit" href="/">
+        Menu Mingle
+      </Link>
+      {new Date().getFullYear()}.
+    </Typography>
+  );
+};
+
+const serverUrl = import.meta.env.VITE_APP_SERVER_URL;
+
+const SignUp = () => {
+  const [role, setRole] = React.useState();
+  const [errors, setErrors] = React.useState({});
```

Comment:
You only use a single error, so name the state as error, not errors.
And the type for any error is `unknown` or `any`, and `any` is preferable. set the type for the state to be any and remove the object brackets argument.